Source/WebKit/GPUProcess/mac/GPUProcessMac.mm

```diff
@@ -111,10 +112,12 @@
 
 void GPUProcess::openDirectoryCacheInvalidated(SandboxExtension::Handle&& handle)
 {
-    AuxiliaryProcess::openDirectoryCacheInvalidated(WTFMove(handle));
+    auto cacheInvalidationHandler = [handle = WTFMove(handle)] () mutable {
+        AuxiliaryProcess::openDirectoryCacheInvalidated(WTFMove(handle));
```

Comment:
My understanding is this function populates the cache, how do we make sure the cache will not be used before the function is called, after this patch delays it?

Reply:
That is a very good point. We actually subscribe to a notification about this in the UI process, and then send a message to the GPU and WebContent process to give them a chance to repopulate the cache. Perhaps dispatching the work to a non main thread will not add much to the overall time it takes to populate the cache? The information in the cache is needed when calling `getpwuid_r`, which we only call on startup from `getHomeDirectory()` in WebKit before we are entering the sandbox. However, there could possibly be other frameworks that are calling this function or other functions related to the home directory of the user after entering the sandbox.

Thanks for reviewing!

Reply:
I expanded the commit message to explain this closer.